graylog2-server/src/main/java/org/graylog/security/SecurityModule.java

```diff
@@ -60,10 +63,11 @@ protected void configure() {
 
         OptionalBinder.newOptionalBinder(binder(), PermissionAndRoleResolver.class)
                 .setDefault().to(DefaultPermissionAndRoleResolver.class);
-
         OptionalBinder.newOptionalBinder(binder(), GranteeService.class)
                 .setDefault().to(DefaultGranteeService.class);
 
+        Multibinder.newSetBinder(binder(), AdditionalGrantsResolver.class).addBinding().to(DefaultAdditionalGrantsResolver.class);
```

Comment:
Instead of binding an empty grants resolver, you can create an empty multibinder. Guice will then inject an empty set.

```suggestion
        Multibinder.newSetBinder(binder(), AdditionalGrantsResolver.class);
```

Reply:
Neat - good to know